index.d.ts

```diff
@@ -68,57 +96,55 @@ declare module "hyperapp" {
   // ---------------------------------------------------------------------------
 
   // A dispatched action handles an event in the context of the current state.
-  type Dispatch<S> = (action: Action<S>, payload?: Payload<any>) => void
+  type Dispatch<S> = (
+    dispatchable: Dispatchable<S>,
+    payload?: any
+  ) => void
+
+  // A dispatchable entity, when processed, causes a state transition.
+  type Dispatchable<S> = S | StateWithEffects<S> | Reaction<S>
 
   // An action transforms existing state and/or wraps another action.
-  type Action<S, P = any> = ActionTransform<S, P> | ActionWithPayload<S, P>
-  type ActionTransform<S, P = any> = (
-    state: State<S>,
-    payload: Payload<P>
-  ) => State<S> | StateWithEffects<S> | Action<S>
-  type ActionWithPayload<S, P> = [ActionTransform<S, P>, Payload<P>]
+  type Reaction<S, P = any> = Action<S, P> | ActionWithPayload<S, P>
```

Comment:
Can just inline this type? `Reaction` feels out of place, as it is not a formal (or even informal) Hyperapp thing.

Reply:
As per the discussion on Discord, yes `Reaction` will get removed.